File: 1512. Number of Good Pairs/Solution.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        counts = [0 for i in range(101)]
        K = 2

        for i in range(len(nums)):
            counts[nums[i]] += 1

        ans = 0

        for i in range(0,101):
            N = counts[i]
            ans += (N * (N-1))//K

        return ans

File: 1512. Number of Good Pairs/Solution.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int[] counts = new int[101];
        int K = 2;

        for(int i = 0; i<nums.length;i++){
            counts[nums[i]]++;
        }

        int ans = 0;        
        for(int i = 0;i<counts.length;i++){
            int n  = counts[i];
            ans += (n * (n - 1)) / K;
        }

        return ans;
    }
}